src/Compilers/Server/VBCSCompilerTests/DesktopBuildClientTests.cs

```diff
@@ -187,7 +187,9 @@ public async Task ConnectToPipe()
                 Assert.False(connection.IsCompleted);
                 cts.Cancel();
                 await Assert.ThrowsAnyAsync<OperationCanceledException>(
+#pragma warning disable VSTHRD003 // Avoid awaiting foreign Tasks
                     async () => await connection.ConfigureAwait(false)).ConfigureAwait(false);
+#pragma warning restore VSTHRD003 // Avoid awaiting foreign Tasks
```

Comment:
How is JTF getting into the VBCSCompiler Unit tests? That seems like a problem.

Reply:
The analyzers work (and are intended to work) outside of JTF scenarios. This specific case is one where the analyzer needs to be updated to recognize the pattern.

Reply:
> Now, doesn't that look a lot simpler? I'd even put it on one line.

It does look simpler. However, that was my first try locally and it didn't address the issue, so I left it as an unrelated follow-up.

Reply:
> The analyzers work (and are intended to work) outside of JTF scenarios

Sure but i'm not sure how applicable it is to the compiler code base. I get the problem it's trying to solve and why this would be fairly applicable to the VS code base. But so far in the compiler it's just causing churn without fixing any bugs. 



---
In reply to: [268088857](https://github.com/dotnet/roslyn/pull/34342#discussion_r268088857) [](ancestors = 268088857)

Reply:
> I get the problem it's trying to solve and why this would be fairly applicable to the VS code base. But so far in the compiler it's just causing churn without fixing any bugs.

Certainly, if you have no thread affinity anywhere in the compiler and you have an analyzer rule to always use `.ConfigureAwait(false)` to avoid accidentally picking up thread affinity, then perhaps this newly activated rule doesn't make sense for that project as it would always be noise.
Though I believe this PR does actually fix one potential bug -- at least Sam changed one use of `Task` to use `JoinableTask`, so it seems like at least for that project it was worthwhile.

Reply:
> Certainly, if you have no thread affinity anywhere in the compiler and you have an analyzer rule to always use .ConfigureAwait(false) to avoid accidentally picking up thread affinity, then perhaps this newly activated rule doesn't make sense for that project as it would always be noise.

For this project at least (the compiler server tests), that's most definitely the correct statement. The compiler server doesn't even have a concept of a main thread. @sharwell why would we be applying these rules to this project? I could understand arguments for other projects, but this seems quite off here.

Reply:
@jasonmalinowski the vs-threading analyzers are intended to support usage in cases that do not have a main thread. I didn't disable them for VBCS because the false positive rate was very low (two instances total) and I'm hoping we can address at least one of them directly in the analyzers.

Reply:
I guess it's still potentially creating a confusing situation if you have somebody new contributing to this part of the repo, especially for new contributors. Heck, the dev manager of the code didn't get it. The tax *there* is vastly higher than the two false positives.

If we don't have a sane ruleset story to do it, maybe this is a 'wait until .editorconfig' since that'll make it easier to apply to the directory.